smack-integration-test/src/main/java/org/jivesoftware/smack/c2s/SimpleXmppConnectionIntegrationTest.java

```diff
@@ -0,0 +1,72 @@
+/**
+ *
+ * Copyright 2020 Aditya Borikar
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smack.c2s;
+
+import java.util.concurrent.TimeoutException;
+
+import org.jivesoftware.smack.StanzaListener;
+import org.jivesoftware.smack.filter.MessageWithBodiesFilter;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smack.packet.Stanza;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.igniterealtime.smack.inttest.annotations.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.util.IntegrationTestRosterUtil;
+import org.igniterealtime.smack.inttest.util.SimpleResultSyncPoint;
+
+import org.jxmpp.jid.EntityBareJid;
+
+public class SimpleXmppConnectionIntegrationTest extends AbstractSmackIntegrationTest {
+
+    public SimpleXmppConnectionIntegrationTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+    }
+
+    @SmackIntegrationTest
+    public void createConnectionTest() throws TimeoutException, Exception {
+        IntegrationTestRosterUtil.ensureBothAccountsAreSubscribedToEachOther(conOne, conTwo, timeout);
+
+        EntityBareJid userTwo = conTwo.getUser().asEntityBareJid();
+
+        final String messageBody = "Hello from the other side!";
```

Comment:
To be extra paranoid, please mix in the integration run test id in the message body, so that it is unique across integration test runs. This is to prevent a stale message from a previous run being considered the right one.

Reply:
"I must've called a thousand times" :P